skore/src/skore/item/cross_validate_item.py

```diff
@@ -0,0 +1,233 @@
+            The creation timestamp in ISO format.
+        updated_at : str
+            The last update timestamp in ISO format.
+        """
+        super().__init__(created_at, updated_at)
+
+        self.cv_results_serialized = cv_results_serialized
+        self.estimator_info = estimator_info
+        self.X_info = X_info
+        self.y_info = y_info
+        self.plot_bytes = plot_bytes
+
+    @classmethod
+    def factory(
+        cls,
+        cv_results: dict,
+        estimator: sklearn.base.BaseEstimator,
+        X: Data,
+        y: Target | None,
+    ) -> CrossValidationItem:
+        """
+        Create a new CrossValidationItem instance.
+
+        Parameters
+        ----------
+        cv_results : dict
+            The dict output of scikit-learn's cross_validate function.
+        estimator : sklearn.base.BaseEstimator,
+            The estimator that was cross-validated.
+        X
+            The data, input of scikit-learn's cross_validation function.
+        y
+            The target, input of scikit-learn's cross_validation function.
+
+        Returns
+        -------
+        CrossValidationItem
+            A new CrossValidationItem instance.
+        """
+        if not isinstance(cv_results, dict):
+            raise TypeError(f"Type '{cv_results.__class__}' is not supported.")
+
+        cv_results_serialized = {}
+        for k, v in cv_results.items():
+            if k == "estimator":
+                continue
+            if k == "indices":
+                cv_results_serialized["indices"] = {
+                    "train": tuple(arr.tolist() for arr in v["train"]),
+                    "test": tuple(arr.tolist() for arr in v["test"]),
+                }
+            if isinstance(v, numpy.ndarray):
+                cv_results_serialized[k] = v.tolist()
+
+        estimator_info = {
+            "name": estimator.__class__.__name__,
+            "params": repr(estimator.get_params()),
+        }
+
+        y_array = y if isinstance(y, numpy.ndarray) else numpy.array(y)
+        y_info = None if y is None else {"hash": _hash_numpy(y_array)}
+
+        X_array = X if isinstance(X, numpy.ndarray) else numpy.array(X)
+        X_info = {
+            "nb_rows": X_array.shape[0],
+            "nb_cols": X_array.shape[1],
+            "hash": _hash_numpy(X_array),
+        }
+
+        plot_bytes = MediaItem.factory(plot_cross_validation(cv_results)).media_bytes
+
+        instance = cls(
+            cv_results_serialized=cv_results_serialized,
+            estimator_info=estimator_info,
+            X_info=X_info,
+            y_info=y_info,
+            plot_bytes=plot_bytes,
+        )
+
+        return instance
+
+    @property
+    def plot(self):
+        """A plot of the cross-validation results."""
+        return altair.Chart.from_json(self.plot_bytes.decode("utf-8"))
+
+    @property
+    def cv_results(self):
+        """The cross-validation results."""
+        return self.cv_results_serialized
```

Comment:
Its not mandatory to do a property, because `cv_results_serialized` is exposed as public member attribute.
Furthermore, its a bit ambiguous that the property `cv_results` points to serialized one.